File: day-4-hashing/long-consecutive-sequence.cpp
// Longest Consecutive Sequence 
#include<bits/stdc++.h>
using namespace std ;
#define fo(i, n) for(int i = 0; i < n; i++)
#define foo(i, k, n) for(int i = k; i < n; i++)
#define deb(x) {cout << #x << " " << x << endl ;}
#define IOS ios::sync_with_stdio(false), cin.tie(0) ;
#define MOD = 1e9+7 ;
const int N = 1e6+7 ;

int main(int argc, char const *argv[]){
    IOS ;

    vector<int> nums{36, 41, 56, 35, 44, 33, 34, 92, 43, 32, 42};
    unordered_set<int> cache ;
    int answer = 0 ;
    for(auto &&x : nums){ cache.insert(x) ; }
    for(auto &&x : nums){
        if(cache.count(x - 1)) {
            int el = x ;
            while(cache.count(el++)) {
                answer = max(answer, el - x) ;
            }
        }
    }
    cout << answer + 1 << endl ;

    return 0;
}

File: day-4-hashing/long-consecutive-sequence.py
# Longest Consecutive Sequence
# nums = [36, 41, 56, 35, 44, 33, 34, 92, 43, 32, 42]
nums = [1, 9, 3, 10, 4, 20, 2]

cache = set()
for x in nums:
    cache.add(x)
answer = 0
for x in nums:
    if x-1 in nums:
        el = x
        a = []
        while el in cache:
            el=el+1
            a.append(answer)
            a.append(el - x)
            answer = max(a)

print(answer + 1)